lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -88,10 +88,12 @@ private function _constructPdoDsn(array $params)
 
         if (isset($params['dbname'])) {
             $dsn .= 'dbname=' . $params['dbname'] . ' ';
+        } elseif (isset($params['default_dbname'])) {
+            $dsn .= 'dbname=' . $params['default_dbname'] . ' ';
         } else {
```

Comment:
I'd say remove this else block entirely, as postgresql has a default behavior, which is to connect to the database matching the username.